### ISSUE TITLE
Syncs on cell, even when that setting is Off

### ISSUE THREAD

**Issue by jfburke3** [mapillary/mapillary_issues#710]
Android 4.2.2, using the 1.14 build Peter put together to fix the camera-mode notification problem.

I will note that I have the "upload pictures in background" setting ENABLED, but I expected that to only take effect while I was on a wifi network because I have Sync on Cell OFF.

I was at a coffee shop, reading on my tablet and taking advantage of their wifi to upload some of the 2,200 Mapillary pictures on my phone (their upload speed is faster than mine at home :-) ).  I left to head home (without first cancelling upload), stopped at the store on the way, got home, pulled my phone out of my pocket and noticed a warning icon on the notification bar.  Turned out to be my data warning alert; Mapillary had uploaded ~500 pictures (1.4Gb) over the cellular network (curse you, fast 4G!).

I'm as close to 100% sure as I can be that the pre-1.0 version I had been using a week ago did not do this--I'm pretty certain that when I lost wifi connection, it would pause.  However, I'm not sure if I ever used the upload-in-background setting then.

Am I misunderstanding how those two settings interact, or is this a bug?

(On the plus side, y'all are going to get a photo sequence through downtown Atlanta much earlier than would otherwise have happened.)

**Comment by jfburke3** [comment]
As a test last night, I tried disabling wifi on my phone, and it continued to upload via the cell network.

**Comment by peterneubauer** [comment]
That is very odd, any chance you can send a log on that?

**Comment by jfburke3** [comment]
Yes, but I'll have to carefully coordinate this one to avoid using too much cell data, so it might be a day or two.  Might take advantage of that new low-res setting to take a few dozen shots for this test.

Just so I'm sure I understand correctly, then:  once I lost my wifi connection, it should have stopped/paused upload (even if it finished the current-picture upload on cell)?

**Comment by jfburke3** [comment]
Additional comment:

It might be that the app is defaulting to an undefined value for upload on cell that is interpreted as On.

I just installed the 1.15 build from the Play store on my phone, after uninstalling your dropbox 1.14 build (Upgrade wasn't an option in Play for some reason), then went into Settings to see what various values were.  Among them, I realized that Sync on cell didn't specify on or off.  I selected it, answered No, and *then* it said "Sync on cell: Off".  (Before it just said "Sync on cell".)

This takes me back to what might have happened to cause the upload-on-cell in the first place.  I'd been using the pre-1.0 build, and upgraded to 1.0 when it was released.  I never could get it to pull down my feed or profile, so after a few days of fiddling with that and getting frustrated, I *uninstalled* 1.0, then reinstalled it.  So *if* it went back to default values then, and it did the same thing as 1.15 just did, then Sync on cell might not be defined.

Somewhere in the course of trying to figure out why it was uploading on the cell network yesterday, I must've actually set the parameter without realizing it, because when I opened this case, it *did* say Off.

**Comment by peterneubauer** [comment]
I did a new build with more debug, can you try https://dl.dropboxusercontent.com/u/5925006/mapillary/releases/app-release_current.apk and send the log if it fails again?

**Comment by jfburke3** [comment]
Installed.  Will look for an opportunity to get a decent sequence for this test.

**Comment by peterneubauer** [comment]
Good. let me know, Want to resolve this before releasing the next version to Play.

**Comment by jfburke3** [comment]
Logs sent, but I'm not sure I followed the same procedure as before, so you might not see the problem.  :-/  Among other things, I started with bluetooth disabled, and it seemed to behave differently.   After enabling it, behavior seemed closer to what I expected but not quite identical. Let me know if you want me to redo the test.

**Comment by peterneubauer** [comment]
Yeah,
if you encounter uploading over cell again, please send the log directly,
there is more logging around this in place now.

**Comment by jfburke3** [comment]
New log sent.  I think it should have at least some upload-on-cell this time.

To get as close as I could to what happened over the weekend, I did these steps:
* Uninstalled Mapillary app.
* rebooted phone
* Installed your extra-logging build
* Captured a sequence of 48 pictures this morning.
* Started upload while on wifi.  Upload in background enabled, Sync when not charging enabled, Sync on cell doesn't say what the setting is.
* Walked out of range of wifi.  At the time, the Notifications screen said 4 of 48.
* Eventually the upload errored out.  When it did, in the app it showed 36 pictures still to go in the Review screen.

So, either it uploaded at least 8 pics on cell, or the Notifications screen is way off.

Please let me know if you need anything else for this.

**Comment by peterneubauer** [comment]
So, I prepared a new version - I think I found the bug! https://dl.dropboxusercontent.com/u/5925006/mapillary/releases/app-release_current.apk and please report back!

**Comment by jfburke3** [comment]
Installed new build. I'll try to geta photo sequence uploaded today using the same steps as yesterday. (Did you want me to use that new trace logs setting or send the normal debug log?)

**Comment by peterneubauer** [comment]
Send the normal log, should work.

Thanks! If it works correctly now, I will release.

**Comment by jfburke3** [comment]
Ok, tested.  The good news is that the problem-as-originally-reported did not recur.

The bad news is that what I would consider a new bug showed up, but I wouldn't call it a showstopper that prevents a release.  I'll send you the logs so you can look at them, but just to document what I saw:

* Began as yesterday (uninstall, fresh install, etc.).  Took some pics.  Enabled Upload automatically, began upload on wifi, drove out of range, and the upload stopped at 14 of 86 (or was it 84? can't remember) pics.  It didn't give any messages, the upload just...hung.  Around 6-8 minutes later, I was within range of a *different* wifi network that I'd already configured my phone to automatically connect to, which it did.  Upload did *not* automatically resume, even though the Upload automatically setting was enabled.  An hour later I returned to my original wifi, and it seems to have uploaded a couple more pictures, but not many, and is now hung at 16 of 80.  Not a new picture set, still the ones I took earlier.

So, it did NOT upload on cell, even though it had what should have been enough time to do so, but it also is having trouble automatically resuming.  Like I said, I wouldn't consider that a showstopper.  I wouldn't even call it a high priority, really.

**Comment by peterneubauer** [comment]
@jfburke3 try v 1.17, https://dl.dropboxusercontent.com/u/5925006/mapillary/releases/app-release_current.apk - trying to cancel sync when detecting leaving cheap network connection.

**Comment by jfburke3** [comment]
Peter, 

Um, not sure what I'm testing here. What is a "cheap network connection"?

Assuming this is related to my additional comments on the cell upload fix, where uploads did not automatically resume when I connected to a different wifi network:

Yes, this build did resume upload when I switched wifi networks, and did not upload over cell. 

By coincidence I was at the same coffee shop when I got your email. I installed your Dropbox build and then continued the upload I'd been doing. I left the coffee shop and drove home, about 15 minutes. I checked the notifications periodically during the drive and the current number never changed.  When I got home my phone automatically connected to my wifi, and after a minute or two I saw the current number begin to increment again. 
So if that's what you were working on, then I'd say you have a winner! 

And if I got it wrong, just let me know and I'll try other tests. 

-jack
 

On March 1, 2015 5:44:02 PM EST, Peter Neubauer <notifications@github.com> wrote:
>@jfburke3 try v 1.17,
>https://dl.dropboxusercontent.com/u/5925006/mapillary/releases/app-release_current.apk
>- trying to cancel sync when detecting leaving cheap network
>connection.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/mapillary/mapillary_issues/issues/710#issuecomment-76635972

-- 
Typos courtesy of fancy auto-spell technology.
